CONTRIBUTING.md

```diff
@@ -106,3 +106,67 @@ To update it, run:
+- `docs` contains the old [BuckleScript
+  manual](https://melange.re/melange/Manual.html) which we host to consult
+  (rarely).
+- `jscomp` is where the compiler implementation and some tests live:
+    - `common` defines the Melange `common` private library, housing code
+      shared by both the Melange binary and the `melange.ppx` library.
+    - `core` defines the `core` Melange library, containing the bulk of the
+      compiler backend implementation.
+    - `ext` contains the sources for the `ext` private library. This is a
+      standard library extension with additional functions and data structutres
+      used throughout the Melange code.
+    - `js_parser` is a vendored copy of the [Flow
+      parser](https://github.com/facebook/flow/tree/main/src/parser), used in
+      Melange to classify `%mel.raw` JavaScript code.
+    - `runtime` is the Melange runtime. It has:
+        - the `melange.js` library.
+        - the `Caml_*` modules, implementing the low-level OCaml primitives.
+          While these are technically part of the `melange.js` library, they:
+            1. aren't exposed in the `Js.*` module.
+            2. are only ever accessed from JS: Melange internals don't address
+               any of these modules directly, and you shouldn't either.
+            3. are still exposed e.g. as `Js__Caml_array` because of Dune
+               namespacing implementation details.
+    - `stdlib` is the Melange stdlib, included in the final distribution.
+      - Depends on the runtime
+    - `others` contains the sources for the additional opt-in libraries
+      distributed with Melange: `melange.belt`, `melange.dom` and
+      `melange.node`.
+    - `test` has both:
+        - the sources for the Melange runtime tests that are executed on every
+          CI run
+        - a snapshot of their compilation to JavaScript (in
+          [`jscomp/test/dist`](https://github.com/melange-re/melange/tree/main/jscomp/test/dist)
+          and
+          [`jscomp/test/dist-es6`](https://github.com/melange-re/melange/tree/main/jscomp/test/dist-es6))
+            - **NOTE**: these snapshots are currently built manually. To do so,
+              comment the only line in
+              [`jscomp/dune`](https://github.com/melange-re/melange/blob/main/jscomp/dune)
+              and run `dune build`.
+- `ocaml-tree` has a code generation tool that produces files such as
+  [`core/js_record_fold.ml`](https://github.com/melange-re/melange/blob/main/jscomp/core/js_record_fold.ml),
+  automating some repetitive traversal tasks of the Melange JavaScript
+  intermediate representation.
+- `ppx` has the code for `melange.ppx`, for example:
+    - all the `%mel.*` extensions and `@deriving` derivers are declared in
+      [`ppx/melange_ppx.ml`](https://github.com/melange-re/melange/blob/main/ppx/melange_ppx.ml).
+    - the Melange FFI (Foreign Function Interface) `external`s and attributes
+      are interpreted in the PPX.
```

Comment:
One part that confuses me about the current setup is that the PPX is not self-contained right? In the sense that part of the interpretation of these externals happens in the PPX, but another part happens at lambda compilation time.

E.g. the ppx inlines here some int64 constants

https://github.com/melange-re/melange/blob/1167ca745c7ddc2b950559e53d2ebe43585f3850/ppx/melange_ppx.ml#L655

that are later picked up by lambda

https://github.com/melange-re/melange/blob/1167ca745c7ddc2b950559e53d2ebe43585f3850/jscomp/core/lam_convert.ml#L544

I am not sure if it's worth mentioning here that `melange.ppx` is not a traditional ppx, in that sense, but is designed to work with the rest of the compiler.